src/Workspaces/Remote/ServiceHub/Shared/RemoteEndPoint.cs

```diff
@@ -99,96 +102,106 @@ public void Dispose()
         public async Task InvokeAsync(string targetName, IReadOnlyList<object> arguments, CancellationToken cancellationToken)
         {
             Contract.ThrowIfFalse(_startedListening);
-            cancellationToken.ThrowIfCancellationRequested();
 
             try
             {
                 await _rpc.InvokeWithCancellationAsync(targetName, arguments?.AsArray(), cancellationToken).ConfigureAwait(false);
             }
             catch (Exception ex) when (ReportUnlessCanceled(ex, cancellationToken))
             {
+                // Remote call may fail with different exception even when our cancellation token is signaled
+                // (e.g. on shutdown if the connection is dropped):
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
how would we get in here i the cancellationToken was triggered.  Wouldn't REportUnlessCancled *not* have us come in here since it *was* canceled?

Reply:
ReportUnlessCanceled always returns true, so the exception is always caught.